pkg/antctl/raw/packetcapture/command.go

```diff
@@ -0,0 +1,337 @@
+	timeout   time.Duration
+	number    int32
+	flow      string
+	outputDir string
+}
+
+var option = &options{}
+
+var packetCaptureExample = `  Start capturing packets from pod1 to pod2, both Pods are in Namespace default
+  $ antctl packetcaputre -S pod1 -D pod2
+  Start capturing packets from pod1 in Namespace ns1 to a destination IP
+  $ antctl packetcapture -S ns1/pod1 -D 192.168.123.123
+  Start capturing UDP packets from pod1 to pod2, with destination port 1234
+  $ antctl packetcapture -S pod1 -D pod2 -f udp,udp_dst=1234
+  Save the packets file to a specified directory
+  $ antctl packetcapture -S 192.168.123.123 -D pod2 -f tcp,tcp_dst=80 -o /tmp
+`
+
+func init() {
+	Command = &cobra.Command{
+		Use:     "packetcapture",
+		Short:   "Start capture packets",
+		Long:    "Start capturing packets on the target flow",
+		Aliases: []string{"pc", "packetcaptures"},
+		Example: packetCaptureExample,
+		RunE:    packetCaptureRunE,
+	}
+
+	Command.Flags().StringVarP(&option.source, "source", "S", "", "source of the the PacketCapture: Namespace/Pod, Pod, or IP")
+	Command.Flags().StringVarP(&option.dest, "destination", "D", "", "destination of the PacketCapture: Namespace/Pod, Pod, or IP")
+	Command.Flags().Int32VarP(&option.number, "number", "n", 0, "target number of packets to capture, the capture will stop when it is reached")
+	Command.Flags().StringVarP(&option.flow, "flow", "f", "", "specify the flow (packet headers) of the PacketCapture, including tcp_src, tcp_dst, udp_src, udp_dst")
+	Command.Flags().BoolVarP(&option.nowait, "nowait", "", false, "if set, command returns without retrieving results")
+	Command.Flags().StringVarP(&option.outputDir, "output-dir", "o", ".", "save the packets file to the target directory")
+}
+
+var protocols = map[string]int32{
+	"icmp": 1,
+	"tcp":  6,
+	"udp":  17,
+}
+
+func getPodFileCopier(cmd *cobra.Command) (raw.PodFileCopier, error) {
+	config, client, _, err := getClients(cmd)
+	if err != nil {
+		return nil, err
+	}
+	return raw.NewPodFileCopier(config, client), nil
+}
+
+func getConfigAndClients(cmd *cobra.Command) (*rest.Config, kubernetes.Interface, antrea.Interface, error) {
+	kubeConfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	k8sClientset, client, err := raw.SetupClients(kubeConfig)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to create clientset: %w", err)
+	}
+	return kubeConfig, k8sClientset, client, nil
+}
+
+func getPCName(src, dest string) string {
+	replace := func(s string) string {
+		return strings.ReplaceAll(s, "/", "-")
+	}
+	prefix := fmt.Sprintf("%s-%s", replace(src), replace(dest))
+	if option.nowait {
+		return prefix
+	}
+	return fmt.Sprintf("%s-%s", prefix, rand.String(8))
+}
+
+func packetCaptureRunE(cmd *cobra.Command, args []string) error {
+	option.timeout, _ = cmd.Flags().GetDuration("timeout")
+	return packetCaptureRun(cmd, option)
+}
+
+func packetCaptureRun(cmd *cobra.Command, option *options) error {
```

Comment:
I think you could easily remove `cmd *cobra.Command` from the parameters here, and replace it with the context, out (`io.Writer`) and the clients. It would simplify testing (no need to build a command object, and no need to overwrite `getClients`)?